pkg/controller/subscription/rhmiConfigs/main.go

```diff
@@ -69,13 +70,60 @@ func UpdateStatus(ctx context.Context, client k8sclient.Client, config *integrea
 		config.Status.Maintenance.Duration = strconv.Itoa(WINDOW) + "hrs"
 	}
 
+	// Calculate the upgrade window
 	if installplan.Spec.Approved {
 		config.Status.Upgrade.Window = ""
 	} else {
 		upStart := installplan.ObjectMeta.CreationTimestamp.Time.Format("2 Jan 2006")
-		upEnd := installplan.ObjectMeta.CreationTimestamp.Time.Add((time.Hour * 24) * 14).Format("2 Jan 2006")
+		upEnd := installplan.ObjectMeta.CreationTimestamp.Time.Add(daysDuration(14)).Format("2 Jan 2006")
 		config.Status.Upgrade.Window = upStart + " - " + upEnd
 	}
+
+	// Calculate the upgrade schedule based on the spec:
+	// * If it's set to AlwaysImmediatly, there's no schedule
+	if config.Spec.Upgrade.AlwaysImmediately {
+		config.Status.Upgrade.Scheduled = nil
+
+	} else
+	// * If it's set to next maintenance, it's scheduled at the next maintenance window
+	if config.Spec.Upgrade.DuringNextMaintenance {
+		mtStart, _ := time.Parse("2-1-2006 15:04", config.Status.Maintenance.ApplyFrom)
+		config.Status.Upgrade.Scheduled = &integreatlyv1alpha1.UpgradeSchedule{
+			For:            mtStart.Format("2 Jan 2006 15:04"),
```

Comment:
```suggestion
			For:            mtStart.Format(integreatlyv1alpha1.DateFormat),
```

I think we can use `DateFormat` constant instead from `rhmiconfig_types.go` so that can change this format in a singular place if we need to. Would be good if this constant is used in the unit tests also
https://github.com/integr8ly/integreatly-operator/blob/917e1c88aae3e8b43a20a45bfb1ff4d7ecd022c1/pkg/apis/integreatly/v1alpha1/rhmiconfig_types.go#L61

Reply:
Good point. And the funny thing is it was me who introduced that constant! :hand_over_mouth: